packages/gatsby/src/redux/nodes.ts

```diff
@@ -354,6 +354,52 @@ export const ensureIndexByQuery = (
   postIndexingMetaSetup(filterCache, op)
 }
 
+export function ensureEmptyFilterCache(
```

Comment:
I'm wondering why would we need to cache it? Should we query it from db every time?

Reply:
I've seen a site that called it repetitively so having it cached makes sense (as we only have this cached in a Map right now, not an array). With the cache it's only a `.slice(0)`, which ought to be cheaper than `[...map.values()]`

Reply:
Fair enough.

Reply:
Oh and consider that the requested node types may be more than one, in which case we don't have a single map for it at all and we'd be repetitively merging multiple maps together. This cache prevents that, too.